src/common/entity/compute_state_display.ts

```diff
@@ -130,6 +131,17 @@ export const computeStateDisplay = (
     }
   }
 
+  if (domain === "update") {
+    // When updating, show "Updating"
+    // When update available, show the version
+    // When update is not available, show "Up-to-date"
+    return compareState === "on"
+      ? isUpdating(stateObj)
+        ? localize("ui.card.update.updating")
```

Comment:
Should we here use the `in_progress` attribute to show `XX%` instead of "Updating" when the entity supports that?

Reply:
I think the attribute % would be cool

Reply:
oh that would be nice. Does the state machine update based on the % attribute?

Reply:
yes

Reply:
> Does the state machine update based on the % attribute?

yes/no. If a number is provided for `in_progress` id does, but `in_progress` can also be a boolean value. In case `true` it is updating, but hasn't got specific progress data, `false` means its not updating.

Reply:
well, in case of not supporting %, it will update when set to true, and then again when set back to false

Reply:
implemented in https://github.com/home-assistant/frontend/pull/12059/commits/9b6cfc742ed841a378c108320fa52c68fe4e8732
preview at the bottom here https://623241cfb5bc6a0008a8fbc1--home-assistant-gallery.netlify.app/#more-info/update

Reply:
Looks really cool! I think it would be visually nice if we show a progress indicator or something while updating. E.g., somewhere separate or on the install button.
Not sure if that all is possible, but I think it would visually help if something is moving/animated in the screen to indicate it is busy doing the update

Reply:
Everything is possible